src/Compilers/Core/Portable/Text/StringBuilderText.cs

```diff
@@ -89,5 +91,45 @@ public override void CopyTo(int sourceIndex, char[] destination, int destination
         {
             _builder.CopyTo(sourceIndex, destination, destinationIndex, count);
         }
+
+        public override void Write(TextWriter textWriter, TextSpan span, CancellationToken cancellationToken = default)
+        {
+#if NETCOREAPP
+            if (span.Start == 0 && span.Length == this.Length)
+            {
+                textWriter.Write(_builder);
+            }
+            else
+            {
+                if (span.Length == 0)
+                    return;
+
+                int chunkOffset = 0;
+
+                foreach (var chunk in _builder.GetChunks())
+                {
+                    var startFromChunk = span.Start - chunkOffset;
+                    if (startFromChunk < chunk.Length)
+                    {
+                        var endFromChunk = span.End - chunkOffset;
+
+                        var startIndex = Math.Max(startFromChunk, 0);
+                        var endIndex = Math.Min(endFromChunk, chunk.Length);
+
+                        textWriter.Write(chunk.Span.Slice(startIndex, endIndex - startIndex));
```

Comment:
```suggestion
                        textWriter.Write(chunk.Span.Slice(startIndex, endIndex - startIndex));
```

is this different from just `chunk.Span[startIndex..endIndex]`?

Reply:
No, I forgot this syntax exists, thanks.

Reply:
Done.